api/lib/infrastructure/repositories/user-repository.js

```diff
@@ -432,6 +455,21 @@ module.exports = {
         }
         throw err;
       });
-  }
+  },
+
+  async updateUserAttributes(id, userAttributes) {
```

Comment:
```suggestion
  async update(user) {
```
Avec dans ton cas user = `{ id: error.meta.userId, samlId: externalUser.samlId }`

Reply:
Le fait de préciser Attributes a pour but de mentionner qu'on peut mettre à jour les attributs direct

Reply:
Question : Faut-il inclure dans cette méthode un _omit_ d'un éventuel `user.Attributes.id` ?